subsys/bluetooth/mesh/lightness_internal.h

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright (c) 2020 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+ */
+/**
+ * @file
+ * @brief Light Lightness model internal defines
+ */
+
+#ifndef LIGHTNESS_INTERNAL_H__
+#define LIGHTNESS_INTERNAL_H__
+
+#include <bluetooth/mesh.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#if defined(CONFIG_BT_MESH_LIGHTNESS_LINEAR)
+#define LIGHT_USER_REPR LINEAR
+#else
+#define LIGHT_USER_REPR ACTUAL
+#endif /* CONFIG_BT_MESH_LIGHTNESS_LINEAR */
+
+/** The lightness server's value is > 0 */
+#define LIGHTNESS_SRV_FLAG_IS_ON BIT(0)
+/** Flag for preventing startup behavior on the server */
+#define LIGHTNESS_SRV_FLAG_NO_START BIT(1)
+
+enum light_repr {
+	ACTUAL,
+	LINEAR,
+};
+
+static inline u32_t lightness_sqrt32(u32_t val)
```

Comment:
Is this something that would be useful in other mesh models? or other NCS modules?

Reply:
I've been wondering the same thing. I've got a similar function for log2 in a different model I'm working on. I'll make a task for upstreaming these into utils or math or something.